### ISSUE TITLE
Is cca_zoo.models.PMD agnostic to the signs of the canonical correlations?

### ISSUE THREAD

**Issue by JohannesWiesner** [jameschapman19/cca_zoo#139]
Hi James,

I am currently implementing a GridSearch to optimize the penalty parameters and the number of components for `cca_zoo.models.PMD`. In order to find the best penalty combinations, I use the mean over the canonical correlations as the score to optimize. I wonder if this is a valid approach, as some of the correlations are negative and others are positive and would effectively cancel out each other? This is basically the same question as [this](https://stats.stackexchange.com/questions/146069/canonical-correlation-analysis-taking-into-account-the-negatieve-correlations) Cross Validated thread. If so, wouldn't it make sense to take the mean of the absolute correlations?

**Comment by jameschapman19** [comment]
I think this question is more about method than code (but I'll highlight where there could be a code problem).

All scores for PMD on the training data should be positive for any dimension.

Scores for the validation data (i.e. the scores used by gridsearch) could indeed be either positive or negative for any given dimension. Therefore if there is a negative score it suggests that the effect is reversed in the validation set and would be a reason to be suspicious of the effect.

With respect to what this means for hyperparameter tuning I'd say absolute values is definitely wrong (it rewards models for finding effects which are the opposite out of sample!). Sum of correlations (I.e. what you have used) or possibly sum of just the correlations greater than zero could work (exclude the negatives).

**Comment by JohannesWiesner** [comment]
Hi James, of course, how could I have not thought about that. Now that I read your answer I feel stupid. Thanks for the answer (and sorry for using GitHub for this, I guess this should have been placed on Cross Validated re-reading my question one more time). 

Greetings,

Johannes

**Comment by jameschapman19** [comment]
In my latest push I have added a sequential module to allow component-wise learning of effects. The idea is to allow us to tune hyperaparameters for each effect separately. I think this is quite an exciting change.

**Comment by jameschapman19** [comment]
Looks something like this:

```
import numpy as np

from cca_zoo.sequential import SequentialModel
from cca_zoo.linear import rCCA
from cca_zoo.data.simulated import LinearSimulatedData
from cca_zoo.model_selection import GridSearchCV

np.random.seed(0)
data = LinearSimulatedData(view_features=[10, 10], latent_dims=5, correlation=0.8)
X, Y = data.sample(200)

rcca = rCCA()
pipe=Pipeline([ ('preprocessing', MultiViewPreprocessing((StandardScaler(), StandardScaler()))),('rcca', rcca) ])
pipe = rcca
param_grid = {
    "c": [0.1, 0.2, 0.3],
}
gs = GridSearchCV(pipe, param_grid, cv=2, verbose=1, n_jobs=1)

model = SequentialModel(
    gs, latent_dimensions=10, permutation_test=True, p_threshold=0.05
)

model.fit([X, Y])
```
